lib/rake/extensiontask.rb

```diff
@@ -242,6 +242,10 @@ def define_native_tasks(for_platform = nil, ruby_ver = RUBY_VERSION, callback =
       # lib_path
       lib_path = lib_dir
 
+      # Update compiled platform/version combinations
+      ruby_versions = (@ruby_versions_per_platform[platf] ||= [])
+      ruby_versions << ruby_ver
```

Comment:
Can we remove similar code in `define_cross_platform_tasks`?

Reply:
I updated and rebased the PR.